File: src/main/java/com/github/kindrat/cassandra/client/util/StringUtil.java
package com.github.kindrat.cassandra.client.util;

import lombok.experimental.UtilityClass;

import java.util.Optional;

@UtilityClass
public class StringUtil {
    public static Optional<String> lastWord(String[] words) {
        return wordAtPosition(words, -1);
    }

    public static Optional<String> wordAtPosition(String[] words, int position) {
        if (position >= 0) {
            return position < words.length ? Optional.of(words[position].trim()) : Optional.empty();
        } else {
            int index = words.length + position;
            return index >= 0 ? Optional.of(words[index].trim()) : Optional.empty();
        }
    }

    public static int findClosingBracketIndex(String inputString) {
        int index = 0;
        int argIndex = 0;

        for (char c : inputString.toCharArray()) {
            if (c == '(') {
                argIndex++;
            }
            if (c == ')') {
                argIndex--;
            }
            if (argIndex == 0) {
                break;
            }
            index++;
        }
        return index;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/util/StringUtil.kt
package com.github.kindrat.cassandra.client.util

object StringUtil {
    @JvmStatic
    fun lastWord(words: Array<String>): String? {
        return wordAtPosition(words, -1)
    }

    fun wordAtPosition(words: Array<String>, position: Int): String? {
        return if (position >= 0) {
            if (position < words.size) words[position].trim { it <= ' ' } else null
        } else {
            val index = words.size + position
            if (index >= 0) words[index].trim { it <= ' ' } else null
        }
    }

    fun findClosingBracketIndex(inputString: String): Int {
        var index = 0
        var argIndex = 0
        for (c in inputString.toCharArray()) {
            if (c == '(') {
                argIndex++
            }
            if (c == ')') {
                argIndex--
            }
            if (argIndex == 0) {
                break
            }
            index++
        }
        return index
    }
}